.circleci/config.yml

```diff
@@ -107,27 +114,48 @@ jobs:
       - run: yarn build:webpack
 
   build-e2e:
+    working_directory: ~/vets-website
     docker:
       - image: cypress/base:10.15.3
     steps:
       - checkout
       - restore_cache:
           keys:
-            - yarn-packages-{{ checksum "yarn.lock" }}
+            - yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          keys:
+            - node-v1-{{ .Branch }}-{{ checksum "package.json" }}
       - run:
-          command: yarn install --frozen-lockfile
+          name: Install Dependencies
+          command: yarn --frozen-lockfile --cache-folder ~/.cache/yarn
+      - run:
+          name: Verify Cypress is Installed
+          command: yarn cypress verify || yarn cypress install
       - save_cache:
-          key: yarn-packages-{{ checksum "yarn.lock" }}
+          key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
           paths:
-            - ~/.cache/yarn
-      - run: mkdir -p ./test-results
-      - run: yarn install-repos
-      - run: yarn fetch-drupal-cache
-      - run: yarn build
+            - ~/.cache
+      - save_cache:
+          key: node-v1-{{ .Branch }}-{{ checksum "package.json" }}
```

Comment:
The [Circle docs](https://circleci.com/docs/2.0/caching/#partial-dependency-caching-strategies) explain the `v1` part:

> Since caches are immutable, this strategy allows you to regenerate all of your caches by incrementing the version.

Reply:
The `v1` is a common naming convention in CircleCI. It allows us to to manually trigger storing new caches if we need to. Not necessary though.

Yes I think that makes sense to use `yarn.lock` for `node_modules` as well. I believe I saw `node_modules` being cached off of `package.json` separately in an example, which is why I did that here. But I don't see any apparent need to do so.

And I think caching `node_modules` in other steps would significantly help. On average it takes Yarn 1 - 2 minutes to install dependencies in this repo. By caching `node_modules`, we wouldn't have reinstall dependencies for every commit on CircleCI. I can make another PR for that.